src/convertToObject.js

```diff
@@ -6,7 +6,18 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map((property) => property.trim())
+    .filter((property) => property !== '')
+    .map((property) => property.split(':').map((element) => element.trim()))
+    .reduce(
+      (object, property) => ({
+        ...object,
+        [property[0]]: property[1],
+      }),
+      {},
+    );
```

Comment:
you used here 4 itteration methods for 1 string its huge performance impact, try to use here only 1 itteration method